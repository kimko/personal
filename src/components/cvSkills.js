import React from "react";
import PropTypes from "prop-types";

export const Skill = (props) => {
  const classn = "f7 b db pa2 link dim ba";
  return (
    <li key={props.skill} className="dib mr1 mb2">
      <div
        className={
          props.main == true ? `${classn}  bg-light-silver` : `${classn} `
        }
      >
        {props.skill}
      </div>
    </li>
  );
};

Skill.propTypes = {
  skill: PropTypes.string,
  main: PropTypes.bool,
};

const processSkillData = (data) => {
  // TODO refactor
  let skills = [];
  let iSkill = 0;
  for (const [skill, value] of Object.entries(data)) {
    iSkill += 100;
    skills.push(<Skill key={skill} main={true} skill={skill} />);
    value.forEach((element, valueIdx) => {
      skills.push(
        <Skill key={iSkill + valueIdx} main={false} skill={element} />,
      );
    });
  }
  return skills;
};

export const Skills = (props) => {
  const skills = processSkillData(props.skillData);
  return (
    <section className="bb w-100 cf pl4" style={props.style}>
      <h1 className="bl bw2 f4 pl1">Technical Skills</h1>
      <ul className="list pl3 pl0-ns pr2">{skills}</ul>
    </section>
  );
};

Skills.propTypes = {
  style: PropTypes.object,
  skillData: PropTypes.object,
};
