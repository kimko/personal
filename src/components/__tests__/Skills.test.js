import React from "react";
import { render, cleanup } from "@testing-library/react";

import { Skills } from "../cvSkills";

afterEach(cleanup);

const skillData = {
  animals: ["cats", "red-pandas"],
  other: ["skill", "data"],
};

it("renders a Skill section", () => {
  const { getByTestId, asFragment } = render(<Skills skillData={skillData} />);

  expect(getByTestId("test-section-skills")).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
