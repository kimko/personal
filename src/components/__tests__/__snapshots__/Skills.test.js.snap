// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Skill section 1`] = `
<DocumentFragment>
  <section
    class="bb w-100 cf pl4"
    data-testid="test-section-skills"
  >
    <h1
      class="bl bw2 f4 pl1"
    >
      Technical Skills
    </h1>
    <ul
      class="list pl3 pl0-ns pr2"
    >
      <li
        class="dib mr1 mb2"
      >
        <div
          class="f7 b db pa2 link dim ba  bg-light-silver"
        >
          animals
        </div>
      </li>
      <li
        class="dib mr1 mb2"
      >
        <div
          class="f7 b db pa2 link dim ba "
        >
          cats
        </div>
      </li>
      <li
        class="dib mr1 mb2"
      >
        <div
          class="f7 b db pa2 link dim ba "
        >
          red-pandas
        </div>
      </li>
      <li
        class="dib mr1 mb2"
      >
        <div
          class="f7 b db pa2 link dim ba  bg-light-silver"
        >
          other
        </div>
      </li>
      <li
        class="dib mr1 mb2"
      >
        <div
          class="f7 b db pa2 link dim ba "
        >
          skill
        </div>
      </li>
      <li
        class="dib mr1 mb2"
      >
        <div
          class="f7 b db pa2 link dim ba "
        >
          data
        </div>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
